server/src/infra/repositories/access.repository.ts

```diff
@@ -112,107 +112,120 @@ export class AccessRepository implements IAccessRepository {
   };
 
   asset = {
-    hasAlbumAccess: (userId: string, assetId: string): Promise<boolean> => {
-      return this.albumRepository.exist({
-        where: [
-          {
-            ownerId: userId,
-            assets: {
-              id: assetId,
-            },
-          },
-          {
-            sharedUsers: {
-              id: userId,
-            },
-            assets: {
-              id: assetId,
-            },
-          },
-          // still part of a live photo is in an album
-          {
+    checkAlbumAccess: async (userId: string, assetIds: Set<string>): Promise<Set<string>> => {
+      if (assetIds.size === 0) {
+        return new Set();
+      }
+
+      return this.albumRepository
+        .createQueryBuilder('album')
+        .innerJoin('album.assets', 'asset')
+        .leftJoin('album.sharedUsers', 'sharedUsers')
+        .select('asset.id', 'assetId')
+        .addSelect('asset.livePhotoVideoId', 'livePhotoVideoId')
+        .where(
+          new Brackets((qb) => {
```

Comment:
Is there a reason you have to use the query builder here?

Reply:
Two main reasons I found to start using query builders:
* Previous queries used an array of `where`s, which worked fine because the entire query was encapsulated in an `EXISTS` check. For some of these complex queries, avoiding false positivies is a lot harder without query builders, if we don't want to go with `UNION ALL` in the SQL (e.g. for each retrieved row, is `livePhotoVideoId` or `assetId` the one that matched the `WHERE` condition.
* As this bulk check can now potentially validate thousands of IDs in a single call, avoiding the ORM to create the full entities (and instead just get raw rows with a small set of selected columns) should reduce memory usage.

Reply:
You can manually control what field are selected even when not using a query builder btw.